events/validators.go

```diff
@@ -138,58 +178,61 @@ func getMaybeSnakeCaseField(gotContext map[string]interface{}, field string) int
 	return nil
 }
 
-func validateCloudEvent(name string, gotBytes, wantBytes []byte) error {
+func validateCloudEvent(name string, gotBytes, wantBytes []byte) *ValidationInfo {
+	vi := &ValidationInfo{
+		Name: name,
+	}
+
 	got := &cloudevents.Event{}
 	err := json.Unmarshal(gotBytes, got)
 	if err != nil {
-		return fmt.Errorf("unmarshalling function-received version of cloud event %q: %v", name, err)
+		vi.Errs = []error{fmt.Errorf("unmarshalling function-received version of cloud event %q: %v", name, err)}
```

Comment:
Append the error if getting an error when unmarshalling `got` and `want` and return if `vi.Errs` is not `nil` so we can check both?